src/azul/logging.py

```diff
@@ -6,9 +6,45 @@
 )
 
 
+class LambdaLoggerFilter(logging.Filter):
+
+    def __init__(self, app: AzulChaliceApp = None):
+        try:
+            # From https://github.com/aws/aws-lambda-base-images/tree/python3.8
+            # Imports global variable present in lambda environment
+            from bootstrap import (  # noqa:AZUL141
+                _GLOBAL_AWS_REQUEST_ID,
+            )
+        except ImportError:
+            # Block exists for local development/lack of aws_request_id variable
+            self.aws_request_id = "local"
+        else:
+            assert app.lambda_context.aws_request_id == _GLOBAL_AWS_REQUEST_ID
+            self.aws_request_id = _GLOBAL_AWS_REQUEST_ID
+
+    def filter(self, record):
+        record.aws_request_id = self.aws_request_id
+        return True
+
+
+log_format = '[%(levelname)s]\t%(asctime)s.%(msecs)03dZ\t%(name)s\t%(aws_request_id)s\t%(message)s\n'
+log_date_format = '%Y-%m-%dT%H:%M:%S'
+
+
 def configure_app_logging(app: AzulChaliceApp, *loggers):
     _configure_log_levels(app.log, *loggers)
 
+    root_logger = logging.getLogger()
+    root_formatter = logging.Formatter(log_format, log_date_format)
+    if len(root_logger.handlers) != 0:
+        root_logger.handlers[0].setFormatter(root_formatter)
+        root_logger.handlers[0].addFilter(LambdaLoggerFilter(app))
+    else:
```

Comment:
Can you add a comment for why this else block is used? It's not apparent why the root_handler would not have a handler configured.

Reply:
In what situations do we expect the else branch to be hit?

Reply:
We should also log in the else branch to indicate that we're using a fallback.

Reply:
Local development e.g. make -C lambdas/service local

Reply:
You didn't actually address this. See my patch.

Reply:
@baladithyab the comment and the log message that were added in the patch are missing from this commit.

From the patch https://github.com/DataBiosphere/azul/pull/3411#pullrequestreview-857444117
```
+        # Otherwise, we're running `chalice local` or unit tests
         handler = logging.StreamHandler()
-        root_logger.info("Initialized default StreamHandler")
-
+        root_logger.info('Root logger has no handlers, must be running ')
```

Reply:
It looks like the logging message in the patch is incomplete, i think it was meant to say `root_logger.info('Root logger has no handlers, must be running locally')`